src/hazard.rs

```diff
@@ -81,8 +81,8 @@ impl<'domain, F> HazardPointer<'domain, F> {
     /// # Safety
     ///
     /// 1. The value loaded from `AtomicPtr` is a valid `&T`, or null.
-    /// 2. The loaded `&T` will only be deallocated through calls to [`HazPtrObject::retire`] on
-    ///    the same [`Domain`] as this holder is associated with.
+    /// 2. The loaded `&T` will only be deallocated through calls to `retire` functions on the same
+    ///    [`Domain`] as this holder is associated with.
```

Comment:
I wasn't 100% sure what to do about links to `HazPtrObject::retire`, since it seems that hasn't existed for a while. Should this link instead to all of the available `retire`/`retire_in` variations?

Reply:
I think like this is fine. I think elsewhere I've also used `retire*` to indicate that there are multiple, but I think what matters here is the general argument, not the specific function.